crates/valence_core/src/text.rs

```diff
@@ -555,6 +555,65 @@ impl Text {
             TextContent::StorageNbt { nbt, .. } => nbt.is_empty(),
         }
     }
+
+    /// Attempts to convert the [`Text`] object to a plain string with the [legacy formatting (`§` and format codes)](https://wiki.vg/Chat#Old_system)
+    ///
+    /// Fails if the [`Text`] object contains anything other than text, color
+    /// and styles, or if any of the colors are not on [the legacy color list](https://wiki.vg/Chat#Colors).
+    pub fn try_into_legacy(&self) -> Option<String> {
```

Comment:
Oops, just now noticed that this is not correct, it doesn't remove formatting (if for example `obfuscated` was set to `Some(false)`). Could probably use a few tests too.

Will take care of this later when I have time. In the meantime waiting for any other comments

Reply:
I don't like this whole thing honestly. I quickly hacked it together until got something that works, but it definitely can be improved. If a color that's not of the 16 valid colors is encountered, instead of failing it could replace it with the closest valid color, for example

Reply:
Considering this is a "best effort" sort of thing I'm OK with the hackyness of it. But some additional improvements and unit tests are welcome :)